packages/config/src/class-textdomain-customizer.php

```diff
@@ -0,0 +1,213 @@
+
+		foreach ( $packages as $package ) {
+			$jetpack_package_prefix = 'automattic/jetpack-';
+			$current_package_prefix = substr( $package->getName(), 0, strlen( $jetpack_package_prefix ) );
+
+			if ( $jetpack_package_prefix !== $current_package_prefix ) {
+				// Not a Jetpack package, so bail.
+				continue;
+			}
+
+			if ( $package->getExtra() && isset( $package->getExtra()['translatable'] ) ) {
+				$files = (array) $package->getExtra()['translatable'];
+				$this->customize_textdomain_in_files( $files );
+			}
+		}
+	}
+
+	/**
+	 * Gets the packages in the local composer repository.
+	 *
+	 * @return array of Composer\Package\PackagesInterface objects.
+	 */
+	protected function get_packages() {
+		$local_repo = $this->composer->getRepositoryManager()->getLocalRepository();
+		return $local_repo->getPackages();
+	}
+
+	/**
+	 * Sets the $vendor_dir instance variable using the vendor directory from
+	 * the composer config.
+	 */
+	protected function set_vendor_dir() {
+		$this->vendor_dir = $this->composer->getConfig()->get( 'vendor-dir' );
+	}
+
+	/**
+	 * Sets the textdomain instance variable using the textdomain provided by
+	 * the plugin. If the plugin did not provide a textdomain, uses 'default'.
+	 */
+	private function set_textdomain() {
+		$root_extra = $this->get_root_extra();
+
+		if ( isset( $root_extra['textdomain'] ) &&
+			is_string( $root_extra['textdomain'] ) ) {
+
+				$this->textdomain = $root_extra['textdomain'];
+				return;
+		}
+
+		$this->textdomain = 'default';
+	}
+
+	/**
+	 * Returns the value of the extra section in the root packages's
+	 * composer.json file.
+	 *
+	 * @return array mixed The root package's extra array.
+	 */
+	protected function get_root_extra() {
+		return $this->composer->getPackage()->getExtra();
+	}
+
+	/**
+	 * Sets the textdomain in the input files and directories.
+	 *
+	 * @param array $files The array of translatable files.
+	 */
+	private function customize_textdomain_in_files( $files ) {
+		if ( ! is_array( $files ) ) {
+			return;
+		}
+
+		foreach ( $files as $file ) {
+			$file_path = $this->vendor_dir . $file;
+
+			if ( ! file_exists( $file_path ) ) {
+				return;
+			}
+
+			if ( is_dir( $file_path ) ) {
+				$iterator = new \RecursiveDirectoryIterator(
+					$file_path,
+					\RecursiveDirectoryIterator::SKIP_DOTS
+				);
+
+				foreach ( new \RecursiveIteratorIterator( $iterator ) as $file_info ) {
+						self::customize_textdomain_in_file( $file_info->getRealPath() );
+				}
+			} elseif ( is_file( $file_path ) ) {
+				self::customize_textdomain_in_file( $file_path );
+			}
+		}
+	}
+
+	/**
+	 * Replaces all occurrences of the placeholder textdomain 'JETPACK_CUSTOMIZE_TEXTDOMAIN'
+	 * in the input file with the plugin's textdomain.
+	 *
+	 * @param string $file_path The file path.
+	 */
+	private function customize_textdomain_in_file( $file_path ) {
+		if ( ! file_exists( $file_path ) ) {
+			return;
+		}
+
+		$file_contents = file_get_contents( $file_path );
```

Comment:
Do you think loading the whole file into memory is going to be scalable? I don't know how to do it line by line yet, I'm sure it's going to be more fuss, do you think it's worth looking into?

Reply:
That’s a good point, and I’ll look into it.

Reply:
Actually, it's going to be *a lot* of fuss with all the file renaming, etc. Let's just keep what we have here and add error handling for file not being writable.